api/src/main/java/gov/cms/ab2d/api/controller/ErrorHandler.java

```diff
@@ -31,7 +31,7 @@
     }
 
     // Most errors that are the fault of the client
-    @ExceptionHandler(IllegalArgumentException.class)
+    @ExceptionHandler({IllegalArgumentException.class, IllegalStateException.class})
```

Comment:
So my concern here is that both these exceptions -- `IllegalArgumentException` and `IllegalStateException` -- are core Java exceptions, which are pretty generic. They can be thrown by other libraries and by Java itself in a variety of situations, to all of which we will react with `BAD_REQUEST`, while in fact the root cause could be something else. Sounds like we need few more our _own_ exceptions.

Say, `InvalidUserInputException` and `InvalidJobStateTransition` exception?